AsyncIO/FileSystem/AsyncFile.cs

```diff
@@ -146,8 +156,10 @@ public static Task MoveAsync(string sourceFileName, string destFileName)
             PathValidator.EnsureCorrectFileSystemPath(sourceFileName);
             PathValidator.EnsureCorrectFileSystemPath(destFileName);
 
+#if !DNXCORE50
             if (Path.GetFullPath(sourceFileName) == Path.GetFullPath(destFileName))
                 return;
+#endif
```

Comment:
Can be replaced by following code without compiler directive: 
            if (Path.Combine(Path.GetDirectoryName(sourceFileName), sourceFileName) == Path.Combine(Path.GetDirectoryName(destFileName), destFileName))
                return;